codewind-che-sidecar/scripts/kube/deployUtils.sh

```diff
@@ -0,0 +1,110 @@
+#!/bin/sh
+
+#*******************************************************************************
+# Copyright (c) 2019 IBM Corporation and others.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v2.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v20.html
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
+#*******************************************************************************
+
+set -o pipefail
+
+SCRIPTS_DIR=/scripts
+DEPLOY_DIR=/tmp/$CHE_WORKSPACE_ID
+DEPLOY_STATUS_FILE=$DEPLOY_DIR/deploy-status.json
+
+function updateDeployStatus() {
+    key=$1
+    file=$2
+    default_message="Failed to update $key in template file." 
+    message=${3:-$default_message}
+    
+    echo "$message Error message appended to $file with key = $key"
+    echo $(jq . $file | jq --arg "$key" "$message" ". + {\"$key\": \"$message\"}") >$file
+}
+
+# setTemplateValue takes in a templated Kubernetes yaml file, and a corresponding key and value to set
+# Uses sed right now. ToDo: Investigate Kustomize
+function setTemplateValue() {
+    yaml_file=$1
+    key=$2
+    value=$3
+
+    sed -i "s/$key/$value/g" $yaml_file
+    exit_code=$?
+    if [[ $exit_code != 0 ]]; then
+        updateDeployStatus $key $DEPLOY_STATUS_FILE
```

Comment:
Are we missing a value arg here?

Reply:
Doesn't seem like it: https://github.com/eclipse/codewind-che-plugin/blob/master/codewind-che-sidecar/scripts/kube/deployPFE.sh#L64

Reply:
Wonder if we're using the default message too much (`Failed to update $key in template file.`)? Maybe we can provide more user-focused messages in a separate issue.

Reply:
I'll get @ssh24 to look at that.